### DISCUSSION TITLE
Problems reading data in ROIs

### DISCUSSION THREAD

**Discussion by Uwe747** [springbok/MLM2PRO-GSPro-Connector#156] [Q&A]
Hi everybody, i´m trying for 2 days now to setup the ROIs. The system always has one or two numbers which are not read correctly. (usually it is the launch angle and the spin axis). I have adjusted the borderlines of the rectangle several times. Even when the numbers are all verified correctly, errors occur within the next strokes. I have no idea what to do. Has anybody a new idea?? greetings

**Comment by wonder99** [comment]
Which launch monitor are you using?  If MLM2PRO, you should really switch to direct BT connection.

**Reply by Uwe747** [reply]
Hi , thanx for your answer.
Yes it is a MLM2Pro. 
BT Connection between which devices??

**Comment by springbok** [comment]
Which launch monitor are you using?Can you upload screenshots of your ROI’s.Sent from my iPhoneOn 13 Dec 2024, at 7:40 am, wonder99 ***@***.***> wrote:﻿
Which launch monitor are you using?  If MLM2PRO, you should really switch to direct BT connection.

**Reply by Uwe747** [reply]
Hi springbok, thank you for your answer. It is the MLM2Pro. Attaches are some pictures 
![ROI1](https://github.com/user-attachments/assets/a11913aa-ab69-42b7-a724-1b9cf8e6b5d4)
![ROI2](https://github.com/user-attachments/assets/558bc25a-e74c-4253-8aba-d48caec970ff)
![ROI3](https://github.com/user-attachments/assets/422d9113-320a-47e8-a262-fb1768805923)

**Reply by Uwe747** [reply]
Even with a smaller rectangle  at the lauch ang. it doesnt read the correct value

**Comment by springbok** [comment]
You need to use the Bluetooth connection, if you look on YouTube there should be how to videos from triple bogey golf and Joe lagowski. In settings change the launch monitor to rapsodo BT, go into the rapsodo app and authorize 3rd party apps which need to be done every 24 hours, on the connector press connect and it should find and connect the the mlm.Sent from my iPhoneOn 13 Dec 2024, at 7:56 am, Uwe747 ***@***.***> wrote:﻿
Even with a smaller rectangle  at the lauch ang. it doesnt read the correct value

**Reply by Uwe747** [reply]
Thank you very much for the tip. The Bluetooth connection worked. However, I often have the issue that when I try to connect to the MLM2Pro, it finds a device called "BlueZ" instead of "MLM-M2P..." and the connection cannot be established. What could be the cause of this?

**Reply by Uwe747** [reply]
Thanx again for your support. 

A short update: we now discover the problem, that we dont get a bluetooth connection to the MLM2PRO. We still have no idea how to solve the problem.
![BT1](https://github.com/user-attachments/assets/81943631-19fb-4b28-a6ba-720601ca6a91)
![BT2](https://github.com/user-attachments/assets/9b490fac-9e38-4fef-bb09-1bc642ae64d6)

Has anyone an idea what we can do??

**Comment by springbok** [comment]
Can you do a troubleshoot firmware install via the rapsodo app, once complete turn the mlm off and then back on, try and connect again.Sent from my iPhoneOn 14 Dec 2024, at 5:39 am, Uwe747 ***@***.***> wrote:﻿
Thanx again for your support.
A short update: we now discover the problem, that we dont get a bluetooth connection to the MLM2PRO. We still have no idea how to solve the problem.
BT1.png (view on web)
BT2.png (view on web)
Has anyone an idea what we can do??

**Reply by Uwe747** [reply]
Good morning Australia, greetings from >Germany :).
We did do troubleshoot. We also turned it off and on. Unfortunately without success.

**Comment by springbok** [comment]
Can you upload your log file in the appdata\logs directory.Sent from my iPhoneOn 14 Dec 2024, at 6:22 am, Uwe747 ***@***.***> wrote:﻿
Good morning Australia, greetings from >Germany :).
We did do troubleshoot. We also turned it off and on. Unfortunately without success.

**Reply by Uwe747** [reply]
Here is the log file
[mlm2pro-gspro-connect.log](https://github.com/user-attachments/files/18131042/mlm2pro-gspro-connect.log)

**Reply by Uwe747** [reply]
Some news, we found that the Connector Software connects to the MLM2PRO when the GSPro Software is closed! (se attached picture). The connector software does not connect to the MLMPro, when GSPro is active
![BT3](https://github.com/user-attachments/assets/5c55bc3e-b55f-4ef2-b647-699e656173aa)

**Comment by Uwe747** [comment]
First of all, thank you very much for your support. We have solved the Bluetooth connection problem. A phone working on the Bluetooth frequency was actually interfering with the connection. We may have a new (hopefully small) problem. But I would open a new thread for that
